cgmes/cgmes-conversion/src/main/java/com/powsybl/cgmes/conversion/elements/dc/AcDcConverterConversion.java

```diff
@@ -0,0 +1,267 @@
+
+        Optional<Boolean> dcTerminalConnected2 = isDcTerminalConnected(converter, TwoSides.TWO, context);
+        boolean defaultConnected2 = getDefaultValue(null, converter.getDcTerminal2().isConnected(), true, true, context);
+        boolean connected2 = dcTerminalConnected2.orElse(defaultConnected2);
+
+        converter.getDcTerminal1().setConnected(connected1);
+        converter.getDcTerminal2().setConnected(connected2);
+    }
+
+    private static void updateActivePowerControl(AcDcConverter<?> converter, PropertyBag cgmesData, Context context) {
+        // Retrieve control mode.
+        AcDcConverter.ControlMode controlMode;
+        String pPccControl = cgmesData.getLocal("pPccControl");
+        if (pPccControl != null && (pPccControl.endsWith("activePower") || pPccControl.endsWith("pPcc"))) {
+            controlMode = AcDcConverter.ControlMode.P_PCC;
+        } else if (pPccControl != null && (pPccControl.endsWith("dcVoltage") || pPccControl.endsWith("udc"))) {
+            controlMode = AcDcConverter.ControlMode.V_DC;
+        } else {
+            boolean isDefaultControlModePpcc = getDefaultValue(null, converter.getControlMode() == AcDcConverter.ControlMode.P_PCC, true, true, context);
+            controlMode = isDefaultControlModePpcc ? AcDcConverter.ControlMode.P_PCC : AcDcConverter.ControlMode.V_DC;
+        }
+
+        if (controlMode == AcDcConverter.ControlMode.P_PCC) {
+            double defaultTargetPpcc = getDefaultValue(null, converter.getTargetP(), 0.0, Double.NaN, context);
+            double targetPpcc = cgmesData.asDouble("targetPpcc");
+            double validTargetPpcc = isValidTargetValue(targetPpcc) ? targetPpcc : defaultTargetPpcc;
+
+            // TargetPpcc must be valid before enabling regulation
+            converter.setTargetP(validTargetPpcc)
+                    .setControlMode(AcDcConverter.ControlMode.P_PCC)
+                    .setTargetVdc(Double.NaN);
+        } else {
+            double defaultTargetVdc = getDefaultValue(null, converter.getTargetVdc(), converter.getDcTerminal1().getDcNode().getNominalV(), Double.NaN, context);
+            double targetVdc = cgmesData.asDouble("targetUdc");
+            double validTargetVdc = isValidTargetV(targetVdc) ? targetVdc : defaultTargetVdc;
+
+            // TargetVdc must be valid before enabling regulation
+            converter.setTargetVdc(validTargetVdc)
+                    .setControlMode(AcDcConverter.ControlMode.V_DC)
+                    .setTargetP(Double.NaN);
+        }
+    }
+
+    private static void updateReactivePowerControl(VoltageSourceConverter vsc, PropertyBag cgmesData, Context context) {
+        String qPccControl = cgmesData.getLocal("qPccControl");
+        if (qPccControl != null && qPccControl.endsWith("voltagePcc")) {
+            double defaultVoltageSetpoint = getDefaultValue(null, vsc.getVoltageSetpoint(), vsc.getTerminal1().getVoltageLevel().getNominalV(), Double.NaN, context);
```

Comment:
Shouldn't we use the nominal voltage of the PCC terminal instead the one of the 1st AC terminal?